File: velosystem-spigot/src/main/java/eu/moonwriters/velosystemspigot/Velosystem.java
package dev.eztxm.velosystem.spigot;

import org.bukkit.plugin.java.JavaPlugin;

public final class VeloSystem extends JavaPlugin {
    private static VeloSystem instance;

    @Override
    public void onEnable() {
        instance = this;
    }

    @Override
    public void onDisable() {
        instance = null;
    }

    public static VeloSystem getInstance() {
        return instance;
    }
}

File: velosystem-spigot/src/main/kotlin/eu/moonwriters/velosystemspigot/VeloSystem.kt
package eu.moonwriters.velosystemspigot

import org.bukkit.plugin.java.JavaPlugin

class VeloSystem : JavaPlugin() {
    override fun onEnable() {
        instance = this
    }

    override fun onDisable() {
        instance = null
    }

    companion object {
        var instance: VeloSystem? = null
            private set
    }
}
